src/util.ts

```diff
@@ -392,17 +391,13 @@ export class Util {
 
         const configWithDefaults: Omit<FinalizedBsConfig, 'rootDir'> = {
             cwd: cwd,
-            deploy: config.deploy === true ? true : false,
             //use default files array from rokuDeploy
             files: config.files ?? [...DefaultFiles],
-            createPackage: config.createPackage === false ? false : true,
             outFile: config.outFile ?? `./out/${rootFolderName}.zip`,
             sourceMap: config.sourceMap === true,
-            username: config.username ?? 'rokudev',
             watch: config.watch === true ? true : false,
             emitFullPaths: config.emitFullPaths === true ? true : false,
-            retainStagingDir: retainStagingDir,
-            copyToStaging: config.copyToStaging === false ? false : true,
```

Comment:
For backwards compatibility (just to make things easier on devs), we should keep `copyToStaging`. Do some type of hierarchy like:
```
 if (options.noEmit is defined){
    use it and ignore copyToStaging
} else if (options.copyToStaging is defined)
    use it
} else {
    set the default
}
```